src/lib/migration-parser.ts

```diff
@@ -57,17 +58,19 @@ const createMigrationParser = function (makeRequest): (migrationCreator: () => a
       return new Entry(apiEntry)
     })
 
-    const ctsWithEntryInfo = await checkEntriesForDeletedCts(intentList, contentTypes, makeRequest)
+    const ctsWithEntryInfo = await fetcher.checkContentTypesForDeletedCts(intentList, contentTypes)
 
     validateChunks(intentList, ctsWithEntryInfo)
 
-    const api = new OfflineAPI(existingCts, entries)
+    const locales = await fetcher.getLocalesForSpace()
+
+    const api = new OfflineAPI(existingCts, entries, locales)
 
     await intentList.compressed().applyTo(api)
 
     const batches = await api.getRequestBatches()
 
-    return batches
+    return { batches }
```

Comment:
This was mainly introduced for the `errorCollector` that we don't have anymore? Should we revert the interface to return only batches?